tests/integration/components/polaris-text-field-test.js

```diff
@@ -0,0 +1,222 @@
+const connectedLeftSelector = buildNestedSelector(connectedContainerSelector, '.Polaris-Connected__Item--connection:first-of-type');
+const connectedRightSelector = buildNestedSelector(connectedContainerSelector, '.Polaris-Connected__Item--connection:last-of-type');
+const connectedPrimaryInputSelector = buildNestedSelector(connectedContainerSelector, '.Polaris-Connected__Item--primary', textFieldSelector);
+const spinnerSelector = '.Polaris-TextField__Spinner';
+const spinnerUpButtonSelector = buildNestedSelector(spinnerSelector, '.Polaris-TextField__Segment:first-of-type');
+const spinnerDownButtonSelector = buildNestedSelector(spinnerSelector, '.Polaris-TextField__Segment:last-of-type');
+const errorSelector = '.Polaris-Labelled__Error';
+const helpTextSelector = '.Polaris-Labelled__HelpText';
+
+const label = 'Text field label';
+const helpText = 'Text field help text';
+const prefix = 'Text field prefix';
+const suffix = 'Text field suffix';
+const left = 'Text field connected left';
+const right = 'Text field connected right';
+const error = 'Text field error message';
+
+module('Integration | Component | polaris-text-field', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders a text input by default', async function(assert) {
+    this.set('label', label);
+
+    await render(hbs`
+      {{polaris-text-field
+        label=label
+      }}
+    `);
+
+    assert.ok(find(textFieldInputSelector), 'it renders a text input');
+    assert.equal(find(labelSelector).textContent.trim(), label, 'it renders the passed-in label');
+  });
+
+  test('when `type` is numeric, it renders a spinner which passes up new values when clicked', async function(assert) {
+    this.setProperties({
+      value: 1,
+      handleChange(num) {
+        this.set('value', num);
+      }
+    });
+
+    await render(hbs`
+      {{polaris-text-field
+        value=value
+        type="number"
+        onChange=(action handleChange)
+      }}
+    `);
+
+    assert.equal(findAll(spinnerSelector).length, 1, 'it renders a polaris-text-field/spinner component');
+
+    await triggerEvent(spinnerUpButtonSelector, 'click');
+    assert.equal(this.get('value'), 2, 'numeric value is incremented when the `up` icon is clicked');
+
+    this.set('value', 5);
+
+    await triggerEvent(spinnerDownButtonSelector, 'click');
+    assert.equal(this.get('value'), 4, 'numeric value is decremented when the `down` icon is clicked');
+  });
+
+  test('it renders as a textarea when `multiline` is true', async function(assert) {
+    await render(hbs`
+      {{polaris-text-field
+        multiline=true
+      }}
+    `);
+
+    assert.equal(findAll(textFieldMultilineSelector).length, 1, 'it renders a textarea');
+  });
+
+  test('it renders an actionable link when a `labelAction` hash is present', async function(assert) {
+    this.setProperties({
+      labelActionText: label,
+      labelActionCalled: false
+    });
+
+    this.set('labelAction', () => {
```

Comment:
Any reason this isn't in the `setProperties` call above?

Reply:
Ah no, added it later on and missed this. Will move.